File: Easy/58_Length_of_Last_Word/Solution.py



class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        ans = ""
        flag = True

        for i in s[::-1]:
            if (i == " " and flag):
                continue
            else:
                flag = False
                if (i != " "):
                    ans += i
                else:
                    break
        
        return len(ans)


if __name__ == "__main__":
    solution = Solution()

    # Test cases 1
    print(solution.lengthOfLastWord("Hello World"))

    # Test cases 2
    print(solution.lengthOfLastWord("   fly me   to   the moon  "))

    # Test cases 3
    print(solution.lengthOfLastWord("luffy is still joyboy"))

File: Easy/58_Length_of_Last_Word/Solution.java


public class Solution {
    public int lengthOfLastWord(String s) {
        String ans = "";
        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) != ' ') {
                ans = s.charAt(i) + ans;
            } else if (!ans.isEmpty()) {
                break;
            }
        }
        return ans.length();
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // Test cases 1
        System.out.println(solution.lengthOfLastWord("Hello World"));

        // Test cases 2
        System.out.println(solution.lengthOfLastWord("   fly me   to   the moon  "));

        // Test cases 3
        System.out.println(solution.lengthOfLastWord("luffy is still joyboy"));
    }
}